### PR TITLE
K8SPSMDB-405 mention strict preserving major versions option in updat…

### PR DESCRIPTION
**Pull request by fiowro**
[K8SPSMDB-405](https://jira.percona.com/browse/K8SPSMDB-405) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

…e doc

### CODE REVIEW

**Comment by delgod** [comment]
maybe it is also needed to mention this behavior in all installation docs.
up to @spron-in

**Review by cap1984 — DISMISSED** [review]

**Comment by fiowro** [comment]
I have checked the installation docs (kubernetes.rst and openshift.rst). We do not describe there _any_ keys from cr.yaml. We even don't discuss different MongoDB versions there. So if we are going to mention some cluster customization options in our installation guides, it's a separate task related to a number of features to be mentioned.

File: source/update.rst [hunk 1]
```diff
@@ -140,6 +140,10 @@ upgraded, the Operator updates the CR to reflect the new image paths and carries
 on sequential Pods deletion in a safe order, allowing StatefulSet to redeploy
 the cluster Pods with the new image.
 
+.. note:: Being enabled, Smart Updates will force the Operator to take MongoDB
```
Comment (spron-in) [inline source/update.rst:143]:
```suggestion
.. note:: Being enabled, Smart Update will force the Operator to take MongoDB
```

**Review by spron-in — CHANGES_REQUESTED** [review]

File: source/update.rst [hunk 2]
```diff
@@ -149,15 +153,22 @@ updates:
...
-     Percona Server for MongoDB 4.2 or 4.0 version
-     regardless of the image path; for already existing clusters, the 4.4
-     vs. 4.2 or 4.0 branch choice will be preserved),
+     Percona Server for MongoDB 4.2, 4.0, or 3.6 version regardless of the image
+     path; for already existing clusters, the 4.4 vs. 4.2, 4.0, or 3.6 branch
+     choice will be preserved),
+   * ``4.4-recommended``, ``4.2-recommended``, ``4.0-recommended``,
+     ``3.6-recommended`` - same as above, but with preserving specific major
+     version in all cases (useful for creating a MongoDB 4.2, 4.0, or 3.6
+     cluster from scratch),
```
Comment (spron-in) [inline source/update.rst:162]:
```suggestion
   * ``4.4-recommended``, ``4.2-recommended``, ``4.0-recommended``,
     ``3.6-recommended`` - same as above, but preserves specific major MongoDB version for newly provisioned clusters (ex. 4.2 will not be upgraded to 4.4 automatically),
```

File: source/update.rst [hunk 3]
```diff
@@ -149,15 +153,22 @@ updates:
...
      the Percona Server for MongoDB 4.4 version will be selected instead of the
-     Percona Server for MongoDB 4.2 or 4.0 version
-     regardless of the image path; for already existing clusters, the 4.4
-     vs. 4.2 or 4.0 branch choice will be preserved),
+     Percona Server for MongoDB 4.2, 4.0, or 3.6 version regardless of the image
+     path; for already existing clusters, the 4.4 vs. 4.2, 4.0, or 3.6 branch
+     choice will be preserved),
+   * ``4.4-latest``, ``4.2-latest``, ``4.0-latest``, ``3.6-latest`` - same as
+     above, but with preserving specific major version in all cases (useful for
+     creating a MongoDB 4.2, 4.0, or 3.6 cluster from scratch),
```
Comment (spron-in) [inline source/update.rst:171]:
```suggestion
   * ``4.4-latest``, ``4.2-latest``, ``4.0-latest``, ``3.6-latest`` - same as above, but preserves specific major MongoDB version for newly provisioned clusters (ex. 4.2 will not be upgraded to 4.4 automatically),
```

File: source/update.rst [hunk 4]
```diff
@@ -149,15 +153,22 @@ updates:
    * ``Recommended`` - automatic upgrades will choose the most recent version
```
Comment (spron-in) [inline source/update.rst:153]:
```suggestion
   * ``Recommended`` - automatic upgrade will choose the most recent version
```

File: source/update.rst [hunk 5]
```diff
@@ -149,15 +153,22 @@ updates:
...
-     regardless of the image path; for already existing clusters, the 4.4
-     vs. 4.2 or 4.0 branch choice will be preserved),
+     Percona Server for MongoDB 4.2, 4.0, or 3.6 version regardless of the image
+     path; for already existing clusters, the 4.4 vs. 4.2, 4.0, or 3.6 branch
+     choice will be preserved),
+   * ``4.4-recommended``, ``4.2-recommended``, ``4.0-recommended``,
+     ``3.6-recommended`` - same as above, but with preserving specific major
+     version in all cases (useful for creating a MongoDB 4.2, 4.0, or 3.6
+     cluster from scratch),
    * ``Latest`` - automatic upgrades will choose the most recent version of
```
Comment (spron-in) [inline source/update.rst:163]:
```suggestion
   * ``Latest`` - automatic upgrade will choose the most recent version of
```

File: source/update.rst [hunk 6]
```diff
@@ -149,15 +153,22 @@ updates:
...
-     vs. 4.2 or 4.0 branch choice will be preserved),
+     Percona Server for MongoDB 4.2, 4.0, or 3.6 version regardless of the image
+     path; for already existing clusters, the 4.4 vs. 4.2, 4.0, or 3.6 branch
+     choice will be preserved),
+   * ``4.4-latest``, ``4.2-latest``, ``4.0-latest``, ``3.6-latest`` - same as
+     above, but with preserving specific major version in all cases (useful for
+     creating a MongoDB 4.2, 4.0, or 3.6 cluster from scratch),
    * *specific version number* - will apply an upgrade if the running Percona
      Server for MongoDB
      version doesn't match the explicit version number with no future upgrades
```
Comment (spron-in) [inline source/update.rst:174]: I don't understand what we are saying here. 
specific version number should run me specific mongoDB version, right?

**Review by spron-in — CHANGES_REQUESTED** [review]

**Review by spron-in — APPROVED** [review]

File: source/update.rst [hunk 7]
```diff
@@ -140,30 +140,39 @@ upgraded, the Operator updates the CR to reflect the new image paths and carries
...
-     version doesn't match the explicit version number with no future upgrades
+     Percona Server for MongoDB 4.2, 4.0, or 3.6 version regardless of the image
+     path; for already existing clusters, the 4.4 vs. 4.2, 4.0, or 3.6 branch
+     choice will be preserved),
+   * ``4.4-latest``, ``4.2-latest``, ``4.0-latest``, ``3.6-latest`` - same as
+     above, but preserves specific major MongoDB version for newly provisioned
+     clusters (ex. 4.4 will not be automatically used instead of 4.2),
+   * *version number* - specify the desired version explicitly
      (version numbers are specified as {{{mongodb44recommended}}},
      {{{mongodb42recommended}}}, etc.),
```
Comment (spron-in) [inline source/update.rst:174]: I'm just curious, if there is a good place where the user can find the versions he can set.

**Review by spron-in — APPROVED** [review]

**Review by cap1984 — APPROVED** [review]

**Pull request merged**